examples/chatbot/actions/sendChat.js

```diff
@@ -0,0 +1,34 @@
+import fakeJAX from '../lib/fakejax'
+import uid from 'uid'
+
+/**
+ * `sendChat` is an ES6 generator. Generators use the `yield` keyword to create
+ * steps in an ES6 iterable.
+ *
+ * Microcosm processes each of these iterations sequentially, using the most
+ * recently resolved iteration as the payload of the action.
+ *
+ * More on generators:
+ * http://babeljs.io/docs/learn-es2015/#generators
+ */
+export default function* sendChat (message) {
```

Comment:
Here's a more concrete example:

```javascript
function* range(start, end) {
  
  while (start <= end) {
    yield start++
  }
  
}


for (let n of range(0, 10)) {
  console.log(n) // 0,1,2,3,4,5,6,7,8,9,10
}
```

Give that a go here: http://babeljs.io/repl/